migration-engine/connectors/sql-migration-connector/src/flavour/postgres.rs

```diff
@@ -558,6 +558,8 @@ where
 
                     if schema_exists_result.get(0).and_then(|row| row.at(1)).and_then(|v| v.to_string()).map(|version| version.contains("CockroachDB")).unwrap_or(false) {
                         circumstances |= Circumstances::IsCockroachDb;
+                        connection.raw_cmd("SET enable_experimental_alter_column_type_general = true").await?;
```

Comment:
The problem with drop&recreate is that it stops working nicely as soon as you have a required column, and we have tests for that. I'll explore doing casts in some cases, since it's a much nicer experience, and cockroach is moving in that direction anyway.

Reply:
Also: we'll still need that query, we're going to add things like the default int size etc. here so we don't rely on server-side cluster settings.